packages/eas-cli/src/build/runBuildAndSubmit.ts

```diff
@@ -587,3 +614,53 @@ async function validateExpoUpdatesInstalledAsProjectDependencyAsync({
     }
   }
 }
+
+function isEnvironment(env: string): env is EnvironmentVariableEnvironment {
+  if (
+    Object.values(EnvironmentVariableEnvironment).includes(env as EnvironmentVariableEnvironment)
+  ) {
+    return true;
+  }
+  return false;
+}
+
+async function resolveEnvVarsAsync({
+  flags,
+  buildProfile,
+  graphqlClient,
+  projectId,
+}: {
+  flags: BuildFlags;
+  buildProfile: ProfileData<'build'>;
+  graphqlClient: ExpoGraphqlClient;
+  projectId: string;
+}): Promise<Record<string, string>> {
+  const environment =
+    flags.environment ?? buildProfile.profile.environment ?? process.env.EAS_CURRENT_ENVIRONMENT;
```

Comment:
I believe that on the backend if you use a `preview`, `development`, or `production` profile it will resolve the corresponding environment for you even if the environment is not specified. Shouldn't we support it here as well?

We don't have to add it now, as we are not doing a full rollout of EAS env vars yet and it's even better that it is not used for real users, but nonetheless it's good to remember to add it at some point to avoid confusion.

Reply:
Maybe we can add linear task for it?